code/modules/mob/living/emote.dm

```diff
@@ -1,344 +1,429 @@
-//This only assumes that the mob has a body and face with at least one mouth.
-//Things like airguitar can be done without arms, and the flap thing makes so little sense it's a keeper.
-//Intended to be called by a higher up emote proc if the requested emote isn't in the custom emotes.
-
-
-
-/mob/living/emote(act, m_type=1, message = null)
-	if(stat == DEAD && (act != "deathgasp") || (status_flags & FAKEDEATH))
-		return
-
-	var/param = null
-
-	if (findtext(act, "-", 1, null)) //Removes dashes for npcs "EMOTE-PLAYERNAME" or something like that, I ain't no AI coder. It's not for players. -Sum99
-		var/t1 = findtext(act, "-", 1, null)
-		param = copytext(act, t1 + 1, length(act) + 1)
-		act = copytext(act, 1, t1)
-
+//The code execution of the emote datum is located at code/datums/emotes.dm
+/mob/living/emote(act, m_type = null, message = null)
 	act = lowertext(act)
-	switch(act)//Hello, how would you like to order? Alphabetically!
-		if ("aflap")
-			if (!src.restrained())
-				message = "<B>[src]</B> flaps [p_their()] wings ANGRILY!"
-				m_type = 2
-
-		if ("blush","blushes")
-			message = "<B>[src]</B> blushes."
-			m_type = 1
-
-		if ("bow","bows")
-			if (!src.buckled)
-				var/M = null
-				if (param)
-					for (var/mob/A in view(1, src))
-						if (param == A.name)
-							M = A
-							break
-				if (!M)
-					param = null
-				if (param)
-					message = "<B>[src]</B> bows to [param]."
+	var/param = message
+	var/custom_param = findchar(act, " ")
+	if(custom_param)
+		param = copytext(act, custom_param + 1, length(act) + 1)
+		act = copytext(act, 1, custom_param)
+
+	var/datum/emote/E = emote_list[act]
+	if(!E)
+		src << "<span class='notice'>Unusable emote '[act]'. Say *help for a list.</span>"
+		return
+	E.run_emote(src, param, m_type)
+
+/* EMOTE DATUMS */
+/datum/emote/living
+	mob_type_allowed_typecache = list(/mob/living)
+	mob_type_blacklist_typecache = list(/mob/living/simple_animal/slime, /mob/living/brain)
+
+/datum/emote/living/aflap

```

Comment:
subtype of flap, so it'll also flap wings
also maybe make the flap speed a var, so this is faster?